streams/src/test/java/org/apache/kafka/streams/kstream/internals/SuppressTopologyTest.java

```diff
@@ -0,0 +1,210 @@
+        "    Processor: KSTREAM-AGGREGATE-0000000002 (stores: [counts])\n" +
+        "      --> myname\n" +
+        "      <-- KSTREAM-SOURCE-0000000005\n" +
+        "    Processor: myname (stores: [])\n" +
+        "      --> KTABLE-TOSTREAM-0000000006\n" +
+        "      <-- KSTREAM-AGGREGATE-0000000002\n" +
+        "    Processor: KTABLE-TOSTREAM-0000000006 (stores: [])\n" +
+        "      --> KSTREAM-MAP-0000000007\n" +
+        "      <-- myname\n" +
+        "    Processor: KSTREAM-MAP-0000000007 (stores: [])\n" +
+        "      --> KSTREAM-SINK-0000000008\n" +
+        "      <-- KTABLE-TOSTREAM-0000000006\n" +
+        "    Sink: KSTREAM-SINK-0000000008 (topic: output-suppressed)\n" +
+        "      <-- KSTREAM-MAP-0000000007\n" +
+        "\n";
+
+    private static final String ANONYMOUS_FINAL_TOPOLOGY = "Topologies:\n" +
+        "   Sub-topology: 0\n" +
+        "    Source: KSTREAM-SOURCE-0000000000 (topics: [input])\n" +
+        "      --> KSTREAM-KEY-SELECT-0000000001\n" +
+        "    Processor: KSTREAM-KEY-SELECT-0000000001 (stores: [])\n" +
+        "      --> KSTREAM-FILTER-0000000004\n" +
+        "      <-- KSTREAM-SOURCE-0000000000\n" +
+        "    Processor: KSTREAM-FILTER-0000000004 (stores: [])\n" +
+        "      --> KSTREAM-SINK-0000000003\n" +
+        "      <-- KSTREAM-KEY-SELECT-0000000001\n" +
+        "    Sink: KSTREAM-SINK-0000000003 (topic: counts-repartition)\n" +
+        "      <-- KSTREAM-FILTER-0000000004\n" +
+        "\n" +
+        "  Sub-topology: 1\n" +
+        "    Source: KSTREAM-SOURCE-0000000005 (topics: [counts-repartition])\n" +
+        "      --> KSTREAM-AGGREGATE-0000000002\n" +
+        "    Processor: KSTREAM-AGGREGATE-0000000002 (stores: [counts])\n" +
+        "      --> KTABLE-SUPPRESS-0000000006\n" +
+        "      <-- KSTREAM-SOURCE-0000000005\n" +
+        "    Processor: KTABLE-SUPPRESS-0000000006 (stores: [])\n" +
+        "      --> KTABLE-TOSTREAM-0000000007\n" +
+        "      <-- KSTREAM-AGGREGATE-0000000002\n" +
+        "    Processor: KTABLE-TOSTREAM-0000000007 (stores: [])\n" +
+        "      --> KSTREAM-MAP-0000000008\n" +
+        "      <-- KTABLE-SUPPRESS-0000000006\n" +
+        "    Processor: KSTREAM-MAP-0000000008 (stores: [])\n" +
+        "      --> KSTREAM-SINK-0000000009\n" +
+        "      <-- KTABLE-TOSTREAM-0000000007\n" +
+        "    Sink: KSTREAM-SINK-0000000009 (topic: output-suppressed)\n" +
+        "      <-- KSTREAM-MAP-0000000008\n" +
+        "\n";
+
+    private static final String NAMED_INTERMEDIATE_TOPOLOGY = "Topologies:\n" +
+        "   Sub-topology: 0\n" +
+        "    Source: KSTREAM-SOURCE-0000000000 (topics: [input])\n" +
+        "      --> KSTREAM-AGGREGATE-0000000002\n" +
+        "    Processor: KSTREAM-AGGREGATE-0000000002 (stores: [KSTREAM-AGGREGATE-STATE-STORE-0000000001])\n" +
+        "      --> asdf\n" +
+        "      <-- KSTREAM-SOURCE-0000000000\n" +
+        "    Processor: asdf (stores: [])\n" +
```

Comment:
Hmm... I think the store name from `InMemoryTimeOrderedKeyValueBuffer.Builder` should show up here as we do

```
if (storeBuilder != null) {
            topologyBuilder.addStateStore(storeBuilder, processorName);
        }
```

in `writeToTopology`, why it does not show up here?

Reply:
ah, it's a regression from the rebase. sorry.

Reply:
Actually, it turns out you were right to begin with. We do need to send the name to the StatefulProcessorNode. The exception I got before was a bug in the StatefulProcessorNode.

Thanks for catching this.